apps/common/models/i-casework.js

```diff
@@ -41,16 +40,19 @@ module.exports = class CaseworkModel extends Model {
   }
 
   save() {
-    const options = this.requestConfig({});
-    options.form = this.prepare();
-    options.method = 'POST';
-    if (!config.icasework.secret || !config.icasework.key && config.env !== 'production') {
-      return Promise.resolve({
-        createcaseresponse: {
-          caseid: 'mock caseid'
-        }
-      });
-    }
-    return this.request(options);
+    return this.prepare().then(formData => {
```

Comment:
Handling the prepare step as async as I said above. Notice there is no token being passed to it. That is because the pdf-upload logic in the prepare step above isn't used if the step itself isn't being overridden by the prepare step in the Compose function. If it is, then this.prepare() calls that first, the token is generated and then passed to the prepare step above.....I told you it was confusing LOL